### PR TITLE
T6678: added ruff lint workflow

### PR DESCRIPTION
**Pull request by kumvijaya**
## Change Summary

T6678: added ruff lint workflow

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Other (please describe):

## Related Task(s)

https://vyos.dev/T6678

## Checklist:

- [x] I have read the [**CONTRIBUTING**](https://github.com/vyos/vyos-1x/blob/current/CONTRIBUTING.md) document
- [ ] I have linked this PR to one or more Phabricator Task(s)
- [x] My commit headlines contain a valid Task id
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

**Review by c-po — APPROVED** [review]

File: .github/workflows/lint-with-ruff.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          ref: ${{ github.event.pull_request.head.ref }}
+          repository: ${{ github.event.pull_request.head.repo.full_name }}
+
+      - name: Setup python
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.11'
```
Comment (zdc) [inline .github/workflows/lint-with-ruff.yml]: Is not this excessive? Python 3.10 included in `ubuntu-latest` seems to be enough.

File: .github/workflows/lint-with-ruff.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+            cat ruff.toml
+          else
+            echo "No ruff.toml found in base repo"
+          fi
+
+      - name: Lint
+        uses: chartboost/ruff-action@491342200cdd1cf4d5132a30ddc546b3b5bc531b
+        with:
+          args: check
+          changed-files: 'true'
```
Comment (zdc) [inline .github/workflows/lint-with-ruff.yml]: Another item with `args: 'format --diff'` is also required.

File: .github/workflows/lint-with-ruff.yml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+      - name: Fetch base ruff config
+        run: |
+          git fetch https://github.com/${{ github.event.pull_request.base.repo.full_name }} ${{ github.event.pull_request.base.ref }}:refs/remotes/origin/base
+          if git ls-tree -r origin/base --name-only | grep -q '^ruff.toml$'; then
+            git checkout origin/base -- ruff.toml
+            echo "Using ruff.toml from base repo"
+            cat ruff.toml
+          else
+            echo "No ruff.toml found in base repo"
+          fi
```
Comment (zdc) [inline .github/workflows/lint-with-ruff.yml]: We can skip this step because it turned out that everything already works as we need by default.

**Review by zdc — CHANGES_REQUESTED** [review]

File: .github/workflows/lint-with-ruff.yml (hunk 1 above)
Comment (c-po) [reply .github/workflows/lint-with-ruff.yml]: VyOS 1.5 uses Python 3.11 so that's a good match using 3.11 here

Comment (kumvijaya) [reply .github/workflows/lint-with-ruff.yml]: Ok, i will keep as 3.11

File: .github/workflows/lint-with-ruff.yml (hunk 3 above)
Comment (kumvijaya) [reply .github/workflows/lint-with-ruff.yml]: ok

File: .github/workflows/lint-with-ruff.yml (hunk 1 above)
Comment (zdc) [reply .github/workflows/lint-with-ruff.yml]: @c-po, sorry, I was not clear enough.

The workflow seems to have no Python dependencies except this single file: https://github.com/ChartBoost/ruff-action/blob/main/action/main.py, and I do not see any reason to update Python.

**Review by dmbaturin — APPROVED** [review]

File: .github/workflows/lint-with-ruff.yml (hunk 1 above)
Comment (kumvijaya) [reply .github/workflows/lint-with-ruff.yml]: @zdc My assumption is that python 3.11 we are setting up here to get the linting/issues for scanned files as per Python 3.11. 
I remember the lint results used to differ as per python version used. I can verify that

Comment (zdc) [reply .github/workflows/lint-with-ruff.yml]: > @zdc My assumption is that python 3.11 we are setting up here to get the linting/issues for scanned files as per Python 3.11. I remember the lint results used to differ as per python version used. I can verify that

Not in this case. Ruff is completely portable and does not need any dependencies, except standard C libraries.

Comment (kumvijaya) [reply .github/workflows/lint-with-ruff.yml]: > > @zdc My assumption is that python 3.11 we are setting up here to get the linting/issues for scanned files as per Python 3.11. I remember the lint results used to differ as per python version used. I can verify that
> 
> Not in this case. Ruff is completely portable and does not need any dependencies, except standard C libraries.

Ok @zdc Removed py version install step

**Review by zdc — APPROVED** [review]

**Pull request merged**